xbmc/video/VideoDatabase.cpp

```diff
@@ -4905,11 +4909,25 @@ void CVideoDatabase::UpdateTables(int iVersion)
     m_pDS->exec("ALTER TABLE settings ADD VideoStream integer");
     m_pDS->exec("ALTER TABLE streamdetails ADD strVideoLanguage text");
   }
+
+  if (iVersion < 104)
+  {
+    m_pDS->exec("ALTER TABLE tvshow ADD duration INTEGER");
+
+    std::string sql = PrepareSQL("SELECT idShow, MAX(c%02d) FROM episode GROUP BY idShow", VIDEODB_ID_EPISODE_RUNTIME);
```

Comment:
I think we can separate the upgrade in 2 queries. The first is something like this:
`SELECT episode.idShow, MAX(episode.c09) from episode LEFT JOIN streamdetails ON episode.idFile = streamdetails.idFile WHERE episode.c09 <> streamdetails.iVideoDuration GROUP BY episode.idShow`
That gathers the runtime set from the scraper (if there is). The second is this query that fills the missing ones with avg/max/latest (whatever we want). In this way we get more reliable results